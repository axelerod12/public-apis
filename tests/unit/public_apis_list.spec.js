import { shallowMount } from '@vue/test-utils';
import FeldmPublicApisList from '@/components/public_apis/public_apis_list.vue';

const apis = [
  { "API": "AdoptAPet", "Description": "Resource to help get pets adopted", "Auth": "apiKey", "HTTPS": true, "Cors": "yes", "Link": "https://www.adoptapet.com/public/apis/pet_list.html", "Category": "Animals" },
  { "API": "Axolotl", "Description": "Collection of axolotl pictures and facts", "Auth": "", "HTTPS": true, "Cors": "no", "Link": "https://theaxolotlapi.netlify.app/", "Category": "Animals" }
];

describe('FeldmPublicApisList', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(FeldmPublicApisList, {
      propsData: {
        apis,
      },
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
