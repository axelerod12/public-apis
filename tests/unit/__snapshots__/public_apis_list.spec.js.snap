// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeldmPublicApisList renders correctly 1`] = `
<div
  class="mx-2 overflow-x-auto"
>
  <table
    class="shadow-lg bg-white w-full"
  >
    <thead>
      <tr>
        <th
          class="bg-blue-100 border text-left p-2"
        >
          API
        </th>
         
        <th
          class="bg-blue-100 border text-left p-2"
        >
          Auth
        </th>
         
        <th
          class="bg-blue-100 border text-left p-2"
        >
          Category
        </th>
         
        <th
          class="bg-blue-100 border text-left p-2"
        >
          Cors
        </th>
         
        <th
          class="bg-blue-100 border text-left p-2"
        >
          Description
        </th>
         
        <th
          class="bg-blue-100 border text-left p-2"
        >
          HTTPS
        </th>
         
        <th
          class="bg-blue-100 border text-left p-2"
        >
          Link
        </th>
      </tr>
    </thead>
     
    <tbody>
      <feldm-public-apis-item-stub
        item="[object Object]"
      />
      <feldm-public-apis-item-stub
        item="[object Object]"
      />
    </tbody>
  </table>
</div>
`;
